packages/core-data/src/hooks/test/use-entity-records.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * WordPress dependencies
+ */
+import triggerFetch from '@wordpress/api-fetch';
+import { createRegistry, RegistryProvider } from '@wordpress/data';
+
+jest.mock( '@wordpress/api-fetch' );
+
+/**
+ * External dependencies
+ */
+import { act, render } from '@testing-library/react';
+
+/**
+ * Internal dependencies
+ */
+import { store as coreDataStore } from '../../index';
+import useEntityRecords from '../use-entity-records';
+
+describe( 'useEntityRecords', () => {
+	let registry;
+	beforeEach( () => {
+		jest.useFakeTimers();
+
+		registry = createRegistry();
+		registry.register( coreDataStore );
+	} );
+
+	afterEach( () => {
+		jest.runOnlyPendingTimers();
+		jest.useRealTimers();
+	} );
+
+	const TEST_RECORDS = [
+		{ id: 1, hello: 'world1' },
+		{ id: 2, hello: 'world2' },
+		{ id: 3, hello: 'world3' },
+	];
+
+	it( 'retrieves the relevant entity record', async () => {
+		let data;
+		await registry
+			.dispatch( coreDataStore )
+			.receiveEntityRecords( 'root', 'widget', TEST_RECORDS, {} );
+		const TestComponent = () => {
+			data = useEntityRecords( 'root', 'widget' );
+			return <div />;
+		};
+		render(
+			<RegistryProvider value={ registry }>
+				<TestComponent />
+			</RegistryProvider>
+		);
+		expect( data ).toEqual( {
+			records: TEST_RECORDS,
```

Comment:
I'm not sure how it works. How is it possible that the data is already there when the resolver still didn't finish its job? Then it gets resolved when time passes. 

I'm sure I'm missing how that works internally but it's a bit confusing when looking at this test alone.